jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/gzip/GzipHandler.java

```diff
@@ -550,14 +551,20 @@ public boolean handle(Request request, Response response, Callback callback) thr
         boolean inflatable = false;
         boolean deflatable = false;
         boolean etagMatches = false;
-        for (HttpField field : fields)
+        boolean seenContentEncoding = false;
+        for (ListIterator<HttpField> i = fields.listIterator(fields.size()); i.hasPrevious();)
         {
+            HttpField field = i.previous();
             HttpHeader header = field.getHeader();
             if (header == null)
                 continue;
             switch (header)
             {
-                case CONTENT_ENCODING -> inflatable = field.contains("gzip");
+                case CONTENT_ENCODING ->
+                {
+                    inflatable = !seenContentEncoding && field.containsLast("gzip");
```

Comment:
If `seenContentEncoding` is true then `inflatable` is set to false.
Is this intended?

Reply:
good catch!